server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -742,6 +742,15 @@ boolean assertSearcherIsWarmedUp(String source, SearcherScope scope) {
     public abstract Translog.Snapshot newChangesSnapshot(String source, MapperService mapperService,
                                                          long fromSeqNo, long toSeqNo, boolean requiredFullRange) throws IOException;
 
+    /**
+     * Creates a new history snapshot from either Lucene/Translog for reading operations whose seqno in the requesting
+     * seqno range (both inclusive).
+     */
+    public Translog.Snapshot newChangesSnapshot(String source, HistorySource historySource, MapperService mapperService,
```

Comment:
Is this useful for anything other than the retention lease pruning used by CCR?

Reply:
Yes, currently used by only CCR